### DISCUSSION TITLE
test case Simultaneous Connect

### DISCUSSION THREAD

**Discussion by seohyeon0322** [ANLAB-KAIST/KENSv3#74] [Project Q&A 2. Transport Layer (TCP)]
1. I found that bind() is also called before connect(). Then, can client socket ip, port #(bound in bind()) be the same as listen socket's? Also, after bind() and connect(), can socket be bound with the address that was used to bind client socket ?

2. I think I failed to understand simultaneous connection synchronization part, and I have some questions about it .

2-1. I thought the socket can get SYN only if there state is LISTEN or SYN_SENT. Is it correct? 

2-2. In the picture, what does '…' means?  It means socket will drop the packet ? or buffer it?

2-3. line 2 and 4 both have seq # 100, I thought it coudn't happend, and in line 5, TCP A expects seq# of 301 but it actually got seq number of 300. I wonder how it can happen.

**Comment by Pusnow** [comment]
1.

> Then, can client socket ip, port #(bound in bind()) be the same as listen socket's?

No, the IP and port number should be client's not server's (listen socket's).

> Also, after bind() and connect(), can socket be bound with the address that was used to bind client socket?

The client will use bound IP and port number to communicate with its peer.

2.

> 2-1. I thought the socket can get SYN only if there state is LISTEN or SYN_SENT. Is it correct?

Yes.

> 2-2. In the picture, what does '…' means? It means socket will drop the packet ? or buffer it?

(I guess this is the picture)

https://github.com/ANLAB-KAIST/KENSv3/blob/22b8748332f44905abb3277dad14fcc088bc36c6/Documentation/rfc/rfc793.txt#L2075-L2093


'...' may mean some delay.


> 2-3. line 2 and 4 both have seq # 100, I thought it coudn't happend, and in line 5, TCP A expects seq# of 301 but it actually got seq number of 300. I wonder how it can happen.

The packet is sent in line 2, and then `TCP B` receives the packet in line 4.
